hawc/apps/lit/api.py

```diff
@@ -424,3 +426,133 @@ def resolve_conflict(self, request, pk):
         )
         instance.resolve_user_tag_conflicts(self.request.user.id, user_reference_tag)
         return Response({"status": "ok"})
+
+    @action(
+        detail=False,
+        url_path=r"search/(?P<id>[\d]+)/type/(?P<type>[\d])",
+        renderer_classes=PandasRenderers,
+        permission_classes=(permissions.IsAdminUser,),
+    )
+    def search(self, request, id: int, type: int):
+        try:
+            type = int(type)
+            id = int(id)
+        except ValueError:
+            raise ValueError(
+                "'type' and 'id' must be integers. Using type '1' for PubMed and '2' for HERO."
+            )
+        choices = (constants.ReferenceDatabase.HERO.value, constants.ReferenceDatabase.PUBMED.value)
+        if type not in choices:
+            raise ValidationError(
+                f"'type' must be either {choices[0]} (for HERO), or {choices[1]} (for PubMed)."
+            )
+
+        model_fields = [
+            "id",
+            "title",
+            "authors_short",
+            "pmid",
+            "hero",
+            "doi",
+            "assessment",
+            "assessment__name",
+            "assessment__dtxsids",
+            "assessment__cas",
+            "published",
+            "study",
+            "study__short_citation",
+            "study__published",
+            "num_robs",
+            "study__bioassay",
+            "study__epi",
+            "study__epi_meta",
+            "study__in_vitro",
+            "study__eco",
+            "num_tags",
+            "num_user_tags",
+        ]
+        column_names = [
+            "HAWC ID",
+            "Title",
+            "Authors",
+            "PubMed IDs",
+            "HERO IDs",
+            "DOIs",
+            "Assessment ID",
+            "Assessment Name",
+            "Assessment DTXSIDs",
+            "Assessment CAS",
+            "Assessment Published",
+            "Study ID",
+            "Study Citation",
+            "Study Published",
+            "Study Risk of Bias Analysis Count",
+            "Study Bioassay",
+            "Study Epidemiological",
+            "Study Epi/Meta",
+            "Study In Vitro",
+            "Study Ecology",
+            "Reference Tags Count",
+            "Reference User Tags Count",
+        ]
+        ids = models.Identifiers.objects.filter(
+            references=OuterRef("pk"),
+            unique_id=id,
+            database=type,
+        )
+
+        qs = (
```

Comment:
can the code that turns a queryset of references and returns a dataframe be moved from here to the lit/managers.py? perhaps as a method on the reference queryset?

If so, I think you could reuse that code for the the reference and the study api